library/src/main/java/com/firebase/ui/auth/core/FirebaseLoginDialog.java

```diff
@@ -92,6 +93,24 @@ public FirebaseLoginDialog setRef(Firebase ref) {
         return this;
     }
 
+    public FirebaseLoginDialog setContext(Context context) {
+        mContext = context;
+        return this;
+    }
+
+    public void reset() {
+        mView.findViewById(R.id.login_section).setVisibility(View.VISIBLE);
+        mView.findViewById(R.id.loading_section).setVisibility(View.GONE);
+    }
+
+    public void logout() {
+        if (mFacebookAuthHelper != null) mFacebookAuthHelper.logout();
```

Comment:
The user of the app authenticated with only one provider. We should only call `logout()` on that provider, not on all of them.